tests/fixtures/database.py

```diff
@@ -928,11 +957,57 @@ def integration_configuration(
         if not isinstance(libraries, list):
             libraries = [libraries]
 
-        integration.libraries.extend(libraries)
+        for library in libraries:
+            if library not in integration.libraries:
+                integration.libraries.append(library)
+
+        if settings is not None:
+            if isinstance(protocol, str):
+                raise ValueError(
+                    "protocol must be an instance of HasIntegrationConfiguration to set settings"
+                )
```

Comment:
The error string is kind of misleading here. I'll update it to: "protocol must be a subclass of HasIntegrationConfiguration to set settings". 

Because `protocol` is a `type`, we can't use `isinstance` to do the check, we can use `issubclass`, but it raises a `TypeError` if you pass it a instance rather then a class (like a string), so we'd have to do a string check anyway.

Reply:
If we wanted to be really explicit here we could do:
```python
if isinstance(protocol, str) or not issubclass(protocol, HasIntegrationConfiguration):
   raise...
```